### ISSUE TITLE
Live price lookup

### ISSUE THREAD

**Issue by cheeseonamonkey** [kinabalu/obsidian-crypto-lookup#2]
I use your plugin and I often want crypto prices on my Obsidian home page. Meaning it would request to the API probably whenever the page loads.

It would be useful and simple enough to implement that I wanted to write an entire plugin from scratch to have it!

But if you wouldn't mind a PR, I think I could take an attempt at contributing to yours.

I'd probably do this by having a keyworded codeblock get replaced with the ticker - similar to how a lot of plugins like DataView, Quoth, or Spotlight do:

![image](https://user-images.githubusercontent.com/54555500/151741571-acfd4ca3-223e-437d-bf28-26f192e1ef7c.png)

I'd probably get the ticker code (_btc_/_eth_/etc) and maybe the vs. price (_usd_) supplied by the user inside the code block like this to be seemless:

![image](https://user-images.githubusercontent.com/54555500/151741487-9b332a28-c641-416a-9ec9-78703ac76946.png)

**Comment by kinabalu** [comment]
PR for the description here is welcome. Happy to review and get it merged
in when you’ve got some thing to show

**Comment by vguillet** [comment]
This feature would be pretty great. Could I add to the implementation request that it would be great to have this data parsable by dataview (as in having the ability to include it as a value that can then be accessed and used in a dataview table as a property). I am currently using obsidian for portfolio management, and the only thing truly missing is live price action to be able to easily compute position values etc... Any thoughts on the possibility of that @chrr

> I use your plugin and I often want crypto prices on my Obsidian home page. Meaning it would request to the API probably whenever the page loads.
> 
> It would be useful and simple enough to implement that I wanted to write an entire plugin from scratch to have it!
> 
> But if you wouldn't mind a PR, I think I could take an attempt at contributing to yours.
> 
> I'd probably do this by having a keyworded codeblock get replaced with the ticker - similar to how a lot of plugins like DataView, Quoth, or Spotlight do:
> 
> ![image](https://user-images.githubusercontent.com/54555500/151741571-acfd4ca3-223e-437d-bf28-26f192e1ef7c.png)
> 
> I'd probably get the ticker code (_btc_/_eth_/etc) and maybe the vs. price (_usd_) supplied by the user inside the code block like this to be seemless:
> 
> ![image](https://user-images.githubusercontent.com/54555500/151741487-9b332a28-c641-416a-9ec9-78703ac76946.png)

**Comment by cheeseonamonkey** [comment]
@vguillet

I agree this would be useful. Maybe even preferred over my original proposition.


It could surely be included in frontmatter tags, although there might be a better way. Any technical ideas would be helpful! 


I've been busy with another project but it is about finished. This is next on my Obsidian wishlist though, I will start it soon :)


I need to do some research as well because I'm not super experienced with Typescript (although I know Javascript well enough). However I have put together a decent encryption plugin already (before discovering that the plugin 'Meld' already existed and worked great lol). It is in my repos, called something like "CryptIt".

**Comment by cheeseonamonkey** [comment]
Making progress:

https://github.com/cheeseonamonkey/obsidian-crypto-lookup/blob/master/main.ts

But now it seems the entire functionality isn't working anymore? Can anyone confirm this?

I think they may have added that Cloudflare DDOS protection, so now when I tried to make the request it just returns that Cloudflare splash page HTML that says to wait 5 Seconds.

It works on a browser because Cloudflare will (after a few seconds) store a cookie. But on both Obsidian and Postman it only receives HTML every time!

Maybe it's also possible it is just a temporary measure while they are experiencing some sort of DDOS attack.

Was this ever an issue before?

**Comment by kinabalu** [comment]
Nope. Must have been something recently added so we’ll either have to find
a new API or figure out how to follow the Cloudflare path till we get a
response

To our success!

Andrew Lombardi

**Comment by cheeseonamonkey** [comment]
I've started a bit of it using the "Nomic" API, it seems pretty good. Let
me see what I can do in the next few days :)

**Comment by vguillet** [comment]
Otherwise have you looked into using the coingeko API? That one should be free with no key requirements, might make it a solid alternative (plus it could allow for fetching a lot more data like market ap, market ap ranking etc ...)

**Comment by cheeseonamonkey** [comment]
I've used CoinGecko's API a bit [here](https://github.com/cheeseonamonkey/CryptoTrckr), the [hosted page](https://mango-flower-0812f0e10.azurestaticapps.net) is still up although its really just something I played with for a couple of days (charts take a bit to load).

The API was great but I ran into some rate-limiting issues on the free version. It was like 50 requests a minute but I remember occasionally going over if I refreshed too often or something. *And 
[their pricing](https://user-images.githubusercontent.com/54555500/153746217-bb6112bf-5ea2-425b-a85a-880409f94c7b.png) is literally laughable.* Its entirely possible I was just using it inefficiently though.

CoinMarketCap has an API too but it is also heavily-monetized if I remember correctly.

Nomics just seemed nice to me after investigating a few because you can get all the ticker data in one request by listing the currencies in the URL:

http://api.nomics.com/v1/currencies/ticker?key=5b7619f704e9f1280cfae0c98813f56e831ee27a&ids=BTC,ETH,XRP,LTC,XMR,BNB,ADA,NOVA,SHIB,DOGE&per-page=100&page=1

Because of this I think the rate limit of 1 per-second is doable even if it gets ugly. Just 1 request to parse into all the currencies.

I just *don't understand* why Cryptonator would put that timed CloudFlare splash screen on their API like that - as far as I can tell, it can only be accessed in a browser by waiting 5 seconds through a splash screen! So apps and Postman just get the HTML of the splash screen as a response. Unless I'm just totally missing something lol.

**Comment by cheeseonamonkey** [comment]
Almost finished, just need to clean up and refactor a bit:

Edit mode:
![image](https://user-images.githubusercontent.com/54555500/153960047-bfca07d0-241c-4e81-99d7-abb354177b06.png)

Preview mode:
![image](https://user-images.githubusercontent.com/54555500/153960020-ab425293-dbb8-4b7a-907d-d0446fabf59b.png)
